redis/src/aio/connection_manager.rs

```diff
@@ -153,15 +207,53 @@ impl ConnectionManager {
         number_of_retries: usize,
         response_timeout: std::time::Duration,
         connection_timeout: std::time::Duration,
+    ) -> RedisResult<Self> {
+        let retry_strategy_info = RetryStrategyInfo::new()
+            .exponent_base(exponent_base)
+            .factor(factor)
+            .number_of_retries(number_of_retries);
+
+        Self::new_with_backoff_and_timeouts_with_max_delay_retry(
+            client,
+            retry_strategy_info,
+            response_timeout,
+            connection_timeout,
+        )
+    }
+
+    /// Connect to the server and store the connection inside the returned `ConnectionManager`.
+    ///
+    /// This requires the `connection-manager` feature, which will also pull in
+    /// the Tokio executor.
+    ///
+    /// In case of reconnection issues, the manager will retry reconnection
+    /// number_of_retries times, with an exponentially increasing delay, calculated as
+    /// rand(0 .. factor * (exponent_base ^ current-try)).
+    ///
+    /// Apply a maximum delay. No retry delay will be longer than this `RetryStrategyInfo.max_delay` .
+    ///
+    /// The new connection will timeout operations after `response_timeout` has passed.
+    /// Each connection attempt to the server will timeout after `connection_timeout`.
+    pub async fn new_with_backoff_and_timeouts_with_max_delay(
```

Comment:
please use a single configuration object that contains both the retry & the timeouts. we don't add many fields, just a single field that represents the whole config of the connection manager.

accordingly, this funciton should be called `new_with_config`

Reply:
Is it possible to name it ConnectionConfigInfo?